azure-pipelines.yml

```diff
@@ -1,44 +1,76 @@
-variables: {
-  MODULE_VERSION: '1.2.2',
-  NODE_8: '8.x',
-  NODE_10: '10.x',
-  NODE_12: '12.x',
-  NODE_14: '14.x'
-}
+variables:
+    { MODULE_VERSION: "1.2.2", NODE_10: "10.x", NODE_12: "12.x", NODE_14: "14.x", NODE_16: "16.x" }
 name: $(MODULE_VERSION)-$(Date:yyyyMMdd)$(Rev:.r)
 
 trigger:
-- main
-- dev
+    branches:
+        include:
+            - main
+            - dev
 
 jobs:
-- job: UnitTests
-  strategy:
-    matrix:
-      WINDOWS_NODE8:
-        IMAGE_TYPE: 'vs2017-win2016'
-        NODE_VERSION: $(NODE_8)
-      WINDOWS_NODE10:
-        IMAGE_TYPE: 'vs2017-win2016'
-        NODE_VERSION: $(NODE_10)
-      WINDOWS_NODE12:
-        IMAGE_TYPE: 'vs2017-win2016'
-        NODE_VERSION: $(NODE_12)
-      WINDOWS_NODE14:
-        IMAGE_TYPE: 'vs2017-win2016'
-        NODE_VERSION: $(NODE_14)
-  pool:
-    vmImage: $(IMAGE_TYPE)
-  steps:
-  - task: NodeTool@0
-    inputs:
-      versionSpec: $(NODE_VERSION)
-    displayName: 'Install Node.js'
-  - script: npm install
-    displayName: 'npm install'
-  - script: npm run test
-    displayName: 'npm build and test'
-    condition: ne(variables['NODE_VERSION'], variables['NODE_8'])
-  - script: npm run test:nolint
-    displayName: 'npm build and test (no linting)'
-    condition: eq(variables['NODE_VERSION'], variables['NODE_8'])
+    - job: UnitTests
+      strategy:
+          matrix:
+              UBUNTU_NODE10:
+                  IMAGE_TYPE: "ubuntu-latest"
+                  NODE_VERSION: $(NODE_10)
+              UBUNTU_NODE14:
+                  IMAGE_TYPE: "ubuntu-latest"
+                  NODE_VERSION: $(NODE_14)
+              UBUNTU_NODE16:
+                  IMAGE_TYPE: "ubuntu-latest"
+                  NODE_VERSION: $(NODE_16)
+              WINDOWS_NODE10:
+                  IMAGE_TYPE: "windows-latest"
+                  NODE_VERSION: $(NODE_10)
+              WINDOWS_NODE14:
+                  IMAGE_TYPE: "windows-latest"
+                  NODE_VERSION: $(NODE_14)
+              WINDOWS_NODE16:
+                  IMAGE_TYPE: "windows-latest"
+                  NODE_VERSION: $(NODE_16)
+              MAC_NODE10:
+                  IMAGE_TYPE: "macOS-latest"
+                  NODE_VERSION: $(NODE_10)
+              MAC_NODE14:
+                  IMAGE_TYPE: "macOS-latest"
+                  NODE_VERSION: $(NODE_14)
+              MAC_NODE16:
+                  IMAGE_TYPE: "macOS-latest"
+                  NODE_VERSION: $(NODE_16)
+      pool:
+          vmImage: $(IMAGE_TYPE)
+      steps:
+          - task: NodeTool@0
+            inputs:
+                versionSpec: $(NODE_VERSION)
+            displayName: "Install Node.js"
+          - script: npm install

```

Comment:
This should be `npm ci` just like the `BuildArtifacts` job. It's very similar to `npm install` but designed to be used in continuous integration systems like azure pipelines (aka why it's called `npm *ci*`)

Reply:
Thanks for the heads up. will do